robot/impl/local_robot_test.go

```diff
@@ -986,7 +987,7 @@ func TestStatus(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 
 		_, err = r.Status(context.Background(), []resource.Name{fail1})
-		test.That(t, err, test.ShouldBeError, errors.Wrapf(errFailed, "failed to get status from %q", fail1))
+		test.That(t, err, test.ShouldBeError, legacyerrors.Wrapf(errFailed, "failed to get status from %q", fail1))
```

Comment:
Does this do something more than `fmt.Errorf` [with a %w](https://pkg.go.dev/fmt#Errorf) to wrap the `errFailed` value?

Reply:
Ah possibly, I didn't investigate this particular failure too closely but will revisit because I think you are correct.

Reply:
It does a bit more - the error is wrapped in a struct internal to `github/pkg/errors`:

```
// Wrapf returns an error annotating err with a stack trace
// at the point Wrapf is called, and the format specifier.
// If err is nil, Wrapf returns nil.
func Wrapf(err error, format string, args ...interface{}) error {
	if err == nil {
		return nil
	}
	err = &withMessage{
		cause: err,
		msg:   fmt.Sprintf(format, args...),
	}
	return &withStack{
		err,
		callers(),
	}
}
```

Reply:
Yeah, it grabs a stack trace, but is that doing anything for this test assertion? I'm uncertain what  `ShouldBeError` does. Does it assert the exact types of the error objects are equal? Does it just assert that `errors.Is(err, legacyerrors.Wrapf(...))`?

Did trying to change to `fmt.Errorf` cause the test to fail?

Reply:
> Did trying to change to fmt.Errorf cause the test to fail?

it did